src/Avalonia.Controls.TreeDataGrid/TreeDataGrid.cs

```diff
@@ -239,6 +240,16 @@ protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
             Scroll = e.NameScope.Find<ScrollViewer>("PART_ScrollViewer");
         }
 
+        protected void OnPreviewKeyDown(object? o, KeyEventArgs e)
+        {
+            if (e.Key == Key.PageDown || e.Key == Key.PageUp)
+            {
+                e.Handled = true;
```

Comment:
I think the handled logic should be moved to the `OnPreviewKeyDown` method, and `e.Handled` set to true only if a scroll could be carried out. Otherwise, this will break the bubbling of the scroll event when a `TreeDataGrid` is nested in a scroll viewer.